packages/server/tests/acceptance/htsPrecompile.spec.ts

```diff
@@ -178,11 +181,20 @@ describe('HTS Precompile Acceptance Tests', async function () {
       const { responseCode } = txBeforeReceipt.events.filter(e => e.event === 'ResponseCode')[0].args;
       expect(responseCode).to.equal(TX_SUCCESS_CODE);
   
-      await baseHTSContract.approvePublic(HTSTokenContractAddress, accounts[2].wallet.address, amount, { gasLimit: 1_000_000 });
-  
-      const txAfter = await baseHTSContract.allowancePublic(HTSTokenContractAddress, BaseHTSContractAddress, accounts[2].wallet.address);
+      await baseHTSContract.approvePublic(HTSTokenContractAddress, accounts[0].wallet.address, amount, { gasLimit: 1_000_000 });
```

Comment:
Is this testing the right flow.
Initially I believe it was right.
`account[0]` approved an allowance of `amount` for `account[2]`
This update is approving for oneself and then sending to someone else, which should work as `accounts[0]` is the treasury no?

Reply:
Yep, you're right. My flow was a wrong. I've fixed it now and left some comments for clarification.